src/sys/socket/addr.rs

```diff
@@ -973,4 +1082,37 @@ pub mod sys_control {
             write!(f, "id: {} unit: {}", self.id(), self.unit())
         }
     }
+
+    impl fmt::Debug for SysControlAddr {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            fmt::Display::fmt(self, f)
+        }
+    }
+}
+
+#[derive(Clone, Copy, Eq, Hash, PartialEq)]
+pub struct EtherAddr(pub u8, pub u8, pub u8, pub u8, pub u8, pub u8);
+
+impl EtherAddr {
```

Comment:
This should have some getters probably to extract the desired info across platforms. Like for your specific use case extracting the address.

Reply:
extracting the address from `EtherAddr` , that's easy:
```rust
    let mac_addr = EtherAddr::new(20,20,20, 20,20,20);
    let a = mac_addr.0;
    let b = mac_addr.1;
    let c = mac_addr.2;
    // ...
```

Reply:
This relates to my point above how the data should not be stored as a tuple, but the backing data store should be a `sockaddr_ll` or `sockaddr_dl` struct. Since this backing store will vary across platforms, having a method on `EtherAddr` that will return just the mac address would be useful.